x-pack/solutions/security/plugins/security_solution/public/management/components/console_argument_selectors/use_custom_scripts_error_toast.tsx

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useEffect } from 'react';
+import { EuiText } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+import type { KibanaReactNotifications } from '@kbn/kibana-react-plugin/public';
+import type { IHttpFetchError } from '@kbn/core/public';
+import type { CustomScriptsErrorType } from '../../hooks/custom_scripts/use_get_custom_scripts';
+
+/**
+ * Shows a danger toast with details if scriptsError is present.
+ * @param scriptsError Error object from custom scripts fetch
+ * @param notifications Kibana notifications service
+ */
+export const useCustomScriptsErrorToast = (
+  scriptsError: IHttpFetchError<CustomScriptsErrorType> | null,
+  notifications: KibanaReactNotifications
+) => {
+  useEffect(() => {
+    if (scriptsError) {
+      let code = 'Error';
+      let message: string | undefined;
+
+      const err = scriptsError;
+      if (err?.body?.message) {
+        const { error } = getMessageFieldFromStringifiedObject(err.body.message) || {};
+        if (error) {
+          code = error.code || code;
+          message = error.message;
+        } else {
+          code = err.body.statusCode ? String(err.body.statusCode) : code;
+          message = err.body.message;
+        }
+      } else {
+        message = err?.message || String(err);
+      }
+
+      if (message) {
+        notifications.toasts.danger({
+          title: code,
+          body: (
+            <EuiText size="s">
+              <p>
+                <FormattedMessage
+                  id="xpack.securitySolution.endpoint.customScripts.fetchError"
+                  defaultMessage="Failed to fetch Microsoft Defender for Endpoint scripts"
+                />
+              </p>
+              <p>{message}</p>
+            </EuiText>
+          ),
+        });
+      }
+    }
+  }, [scriptsError, notifications]);
+};
+export function getMessageFieldFromStringifiedObject(
```

Comment:
I saw your response on this function but I'm still a bit confused and have some concerns around what this function is doing:

1. So this seems to be specific to MS, but this component is suppose to be reusable across EDRs. So how will that work?
2. The logic seems to be parsing a string... then assumes that there is a JSON object in there somewhere and it then `JSON.parse()`'s it into memory - are you sure this is safe?  I feel like this is a security concern: to be "eval"ing unknown code into the browser's memory.

Reply:
Hmm, thanks - your concerns are totally valid :
1. Yes, that's MDE specific, good call - I'll think on how and where to move it
2. You got it right, it's a stringified JSON. Well the only reason was not to provide the whole message because it's soo long, not sure if we should expect malice coming from MDE error, but in general you're right. 

Do you think we should render the whole thing then, or take any other approach?

Reply:
> Re: Do you think we should render the whole thing then, or take any other approach?

Yes. I do not think you should be parsing string values into memory with assumptions that may not hold true in the future.

Maybe the picker can have the "simple" message of "unable to load" and the toast message should display the error as is. I do strongly suggest that you not parse these external messages into JS.

Reply:
Sounds good!